update collectd data model and add ovirt data model

sradco opened the pull request:
Signed-off-by: Shirly Radco <sradco@redhat.com>

t0ffel commented on namespaces/collectd.yml:
```diff
@@ -51,29 +66,54 @@ namespace:
 
     - name: ps_vm
-      type: integer
+      type: group
```
Why do you need to nest value?

sradco replied:
This is the way the nest plugin creates the record.
If there is only 1 value in the record sent by collectd the dsname is ["values"]
If there is more than 1 then the dsname will be something like ["read","write"] and other.

portante replied:
@sradco, let me see if I can clarify.  Here is sample input JSON from `collectd`:

```
{
  "values": [ 1019961344 ],
  "dstypes": [ "gauge" ],
  "dsnames": [ "value" ],
  "interval": 10.0,
  "host": "dhcp-0-135.tlv.redhat.com",
  "plugin": "processes",
  "plugin_instance": "collectd",
  "type": "ps_vm",
  "type_instance": ""
}
```

Then would the normalized form suggested by this commit be:

```
{
  "hostname": "dhcp-0-135.tlv.redhat.com",
  "collectd": {
    "ps_vm": {
      "value": 1019961344
    },
    "dstypes": [ "gauge" ],
    "interval": 10.0,
    "plugin": "processes",
    "plugin_instance": "collectd"
  }
}
```

If so, adding a field name `value` seems unnecessary.  Could you consider this?

```
{
  "hostname": "dhcp-0-135.tlv.redhat.com",
  "collectd": {
    "ps_vm": 1019961344,
    "dstype": "gauge",
    "interval": 10.0,
    "plugin": "processes",
    "plugin_instance": "collectd"
  }
}
```

sradco replied:
I understand that and we should open an issue for the nest plugin to set this differently. 
But until we have this fixed there with the new plugin builds, all in place, I need to have the data model this way.
The plugin is already built and we plan to use it in the next version so the template should match what we have for now.

https://github.com/ViaQ/fluent-plugin-collectd-nest/issues/5

t0ffel commented on namespaces/collectd.yml:
```diff
@@ -212,14 +252,19 @@ namespace:
 
     - name: pending_operations
-      type: integer
+      type:
```
empty type

sradco replied:
will fix

t0ffel commented on namespaces/ovirt.yml:
```diff
@@ -0,0 +1,27 @@
+      The oVirt host UUID.
+
+  - name: engine
+    type: group
```
why is it a group if you have just one value?

sradco replied:
Since I'm thinking that we might add other engine metadata in the future.
What do you suggest?

portante replied:
Why not add other engine field names later?  This is a schema of sorts, but does not record all the fields that have to be there, or all the field names that can only be there.

The purpose is to define the data type of a field when it *_is_* present.  So if new fields show up later, we can add definitions here as necessary.  It is "future proof" by the fact that new fields can just be added into the hierarchy namespace of `ovirt.engine...` whenever necessary.  Does that makes sense?

sradco replied:
If we add new field that is related to oVirt engine I would like to have it under "engine" name space.

Just like we nest the fields under collectd.
We could have just have fields names that have dots between each part but we decided to nest them so if they are under the same plugin for example they will be nested all under that plugin name.

t0ffel commented on namespaces/collectd.yml:
```diff
@@ -301,50 +341,171 @@ namespace:
     type: group
     description: >
-      Corresponds to collectd's CPU plugin.
+      Corresponds to collectd's netlink plugin.
```
is it still netlink plugin?

sradco replied:
No. Will fix

portante commented on namespaces/collectd.yml:
```diff
@@ -1,39 +1,54 @@
     type: string
     description: >
-      Virtual Machine name is in plugin_instance for virt plugin records.
+      Collectd's plugin.
```
Perhaps, "collectd's plugin name"?

sradco replied:
Its the same convention as the other descriptions, like interval, plugin_instance, type etc.
I don't think its critical. I can update if needed.

sradco commented:
I added a patch to fix the nest plugin
https://github.com/ViaQ/fluent-plugin-collectd-nest/pull/6

I fixed the data model according to it.

Please review and merge the nest plugin update and build it
or else the data model will not fit.

portante commented on namespaces/collectd.yml:
```diff
@@ -146,11 +156,6 @@ namespace:
       Corresponds to collectd's disk plugin.
       Or virt plugin with types:disk_octets/disk_ops.
```
Is this statement still true?

portante commented on namespaces/collectd.yml:
```diff
@@ -146,11 +156,6 @@ namespace:
       Corresponds to collectd's disk plugin.
       Or virt plugin with types:disk_octets/disk_ops.
     fields:
-    - name: disk_name
```
Why are we dropping the name field?

sradco replied:
This requires further manipulation for each plugin separately.
I think it will be best to leave as is in the "Plugin_instance" field.
Same for all the "names" you mentioned in the next comments.

portante replied:
Could we consider this from the consumer of the metrics?  If a person is trying to develop graphs or build queries, and they don't know `collectd`, wouldn't it be somewhat confusing to have to use the plugin instance field instead of an explicit name in the same metric subgroup?

Can we drop the `Plugin_instance` field entirely, and move its values to an appropriate identifier fields in each of the metrics subgroups?

portante commented on namespaces/collectd.yml:
```diff
@@ -231,20 +236,15 @@ namespace:
-      Corresponds to collectd's netlink plugin.
+      Corresponds to collectd's interface plugin.
     fields:
-    - name: network_name
```
Why are we dropping the name field?

portante commented on namespaces/collectd.yml:
```diff
@@ -326,18 +426,9 @@ namespace:
     description: >
       Corresponds to collectd's df plugin.
     fields:
-    - name: space_type
```
Why are we dropping these two fields, `space_type` and `mounted_volume`?

sradco replied:
The "space type" is in the "type_instance" and also in the value field nested name.
mounted_volume - same as I commented before, this is plugin specific and I think we should leave as id in plugin_instance.

portante replied:
As for `Plugin_instance`, let's consider moving `type_instance` to the particular subgroup.

sradco replied:
Moved this to the mailing list for discussion.

sradco commented:
What is the status of this ? Can't we merge it? @richm @t0ffel
